modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/utils/OpenApiUtil.java

```diff
@@ -107,23 +115,59 @@ public String getOpenApi(RegisterServiceEntity registerServiceEntity) {
      *
      * @return serviceUrl
      */
-    private String getServiceUrl() {
-        String ip = "localhost";
+    public String getServiceUrl() {
```

Comment:
@baixinsui I'm a little confused about this method. 
If I understood the framework right, it automatically generates the correct URLs and we need not do anything extra. Are the URLs not generated correctly out of the box?

Reply:
Is it ok now?

Reply:
I had used WebMvcLinkBuilder.linkTo(
                    WebMvcLinkBuilder.methodOn(OrchestratorApi.class).openApi(id)).withRel("openApi"); happens
org.springframework.aop.framework.AopConfigException: Could not generate CGLIB subclass of class java.lang.String: Common causes of this problem include using a final class or a non-visible class.                                                                               
I had tried to fix it but not succeed.  And this mothed needs the openApi mothed returns String and only used in OrchestratorApi.class. So that, I use Link.of() and get the baseUrl

Reply:
I use WebMvcLinkBuilder.linkTo(WebMvcLinkBuilder.methodOn(OrchestratorApi.class).openApi(serviceEntity.getId().toString())).withRel("openApi") to fill openApiUrl. This method is used to fill the serviceUrl in html file.